File: app/src/main/java/com/sevtinge/cemiuiler/module/systemui/plugin/HideMiPlayEntry.java
package com.sevtinge.cemiuiler.module.systemui.plugin;

import static com.sevtinge.cemiuiler.module.base.BaseHook.mPrefsMap;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedHelpers;

public class HideMiPlayEntry {
    public static void initHideMiPlayEntry(ClassLoader classLoader) {
        if (mPrefsMap.getStringAsInt("system_ui_control_center_mi_play_entry", 0) == 1) {
            XposedHelpers.findAndHookMethod("miui.systemui.controlcenter.panel.main.external.MiPlayEntryController", classLoader, "available", boolean.class, XC_MethodReplacement.returnConstant(true));
        } else if (mPrefsMap.getStringAsInt("system_ui_control_center_mi_play_entry", 0) == 2) {
            XposedHelpers.findAndHookMethod("miui.systemui.controlcenter.panel.main.external.MiPlayEntryController", classLoader, "available", boolean.class, XC_MethodReplacement.returnConstant(false));
        }
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/systemui/HideMiPlayEntry.kt
package com.sevtinge.cemiuiler.module.systemui

import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.HookUtils
import de.robv.android.xposed.XC_MethodReplacement
import de.robv.android.xposed.XposedBridge

object HideMiPlayEntry : BaseHook() {
    override fun init() {
        val MiPlayPluginManagerClass =
            findClassIfExists("com.android.systemui.controlcenter.phone.controls.MiPlayPluginManager")
        XposedBridge.hookAllMethods(
            MiPlayPluginManagerClass,
            "supportMiPlayAudio",
            object : XC_MethodReplacement() {
                override fun replaceHookedMethod(param: MethodHookParam) {
                    param.result = false
                }
            })

    }
}